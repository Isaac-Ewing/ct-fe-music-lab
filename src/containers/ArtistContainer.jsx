import React, { useState } from 'react';
import ArtistList from '../components/ArtistList';
import Search from '../components/Search';
import { fetchArtists } from '../services/fetchArtists';

const ArtistContainer = () => {
  const [loading, setLoading] = useState(false);
  const [artists, setArtists] = useState([]);
  const [searchTerm, setSearch] = useState('');

  const handleSearch = async (event) => {
    event.preventDefault();
    const artists = await fetchArtists(searchTerm);
    console.log(artists);
    setArtists(artists);
  };

  const handleOtherSearch = ({ target }) => {
    setSearch(target.value);
  };

  return (
    <>
      <Search
        onSubmit={handleSearch}
        onChange={handleOtherSearch}
        searchTerm={searchTerm}
      />
      
      {artists.length ? <ArtistList artists={artists} loading={loading} /> : 0}
    </>
  );
};

export default ArtistContainer;
