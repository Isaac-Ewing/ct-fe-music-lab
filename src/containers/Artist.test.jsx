import React from 'react';
import ArtistContainer from './ArtistContainer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import data from '../fixtures/data.json';

const mockServer = setupServer(
  rest.get('http://musicbrainz.org/ws/2/artist', (req, res, ctx) => {
    return res(ctx.json(data));
  })
);

describe('artist container test', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('the actaul test', async () => {
    const { container } = render(
      <MemoryRouter>
        <ArtistContainer />
      </MemoryRouter>);
    const input = screen.getByRole('textbox');
    const button = screen.getByRole('button', { name: 'search-button' });
    userEvent.type(input, 'killers');
    userEvent.click(button);
    return waitFor(() => {
      const artistList = screen.getAllByText('killers', { exact: false });
      expect(container).toMatchSnapshot();
      expect(artistList.length).toBeGreaterThan(2);
    });
  });
});
