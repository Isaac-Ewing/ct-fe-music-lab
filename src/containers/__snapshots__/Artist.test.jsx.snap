// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artist container test the actaul test 1`] = `
<div>
  <form>
    <label>
      Find an Artist:
      <input
        name="searchTerm"
        type="text"
        value="killers"
      />
    </label>
    <button
      aria-label="search-button"
    >
      Submit
    </button>
  </form>
  <ul
    aria-label="artists"
    name="artists"
  >
    <li>
      <a
        href="/Killers/0dc08b92-4639-4370-94cf-1cd31b0811f6"
      >
        Killers
      </a>
      <h3>
        100
      </h3>
    </li>
    <li>
      <a
        href="/The Killers/95e1ead9-4d31-4808-a7ac-32c3614c116b"
      >
        The Killers
      </a>
      <h3>
        97
      </h3>
    </li>
    <li>
      <a
        href="/Killers/66101a13-70cc-40a7-8371-abcb78012aa3"
      >
        Killers
      </a>
      <h3>
        96
      </h3>
    </li>
    <li>
      <a
        href="/KILLERS/386b27fb-fc2b-400c-a296-079b13e21061"
      >
        KILLERS
      </a>
      <h3>
        93
      </h3>
    </li>
    <li>
      <a
        href="/Killers/1371a633-7446-44c1-8ab9-4e643d7b1832"
      >
        Killers
      </a>
      <h3>
        92
      </h3>
    </li>
    <li>
      <a
        href="/The Killers/c2058349-c474-411b-b857-2bbfc10de344"
      >
        The Killers
      </a>
      <h3>
        85
      </h3>
    </li>
    <li>
      <a
        href="/Studio Killers/cc7d0735-ad9b-4bf9-bfb6-e7bd90945c45"
      >
        Studio Killers
      </a>
      <h3>
        84
      </h3>
    </li>
    <li>
      <a
        href="/Buffalo Killers/7d07fa7b-5019-4c13-861e-66424c100c37"
      >
        Buffalo Killers
      </a>
      <h3>
        82
      </h3>
    </li>
    <li>
      <a
        href="/Cowboy Killers/a30cd414-a9a6-4aab-a302-a3084cc48ff2"
      >
        Cowboy Killers
      </a>
      <h3>
        82
      </h3>
    </li>
    <li>
      <a
        href="/Gasteropodes Killers/df84a4ca-8560-46c9-b01a-283407cf00e0"
      >
        Gasteropodes Killers
      </a>
      <h3>
        81
      </h3>
    </li>
  </ul>
  <button
    disabled=""
  >
    &lt;
  </button>
  <button>
    &gt;
  </button>
</div>
`;
